spock-core/src/main/java/org/spockframework/tempdir/TempDirExtension.java

```diff
@@ -0,0 +1,13 @@
+package org.spockframework.tempdir;
+
+import org.spockframework.runtime.extension.AbstractGlobalExtension;
+
+/**
+ * @author: dqyuan
+ * @date: 2020/07/26
+ */
+public class TempDirExtension extends AbstractGlobalExtension {
+
+  TempDirConfiguration tempDirConfiguration;
```

Comment:
This is not necessary anymore, see http://spockframework.org/spock/docs/2.0-SNAPSHOT/extensions.html#_configuration_objects

You only need to add the config object class to `META-INF/services/spock.config.ConfigurationObject`